packages/web-components/src/components/masthead/masthead-menu-button.ts

```diff
@@ -24,12 +29,49 @@ const { stablePrefix: ddsPrefix } = ddsSettings;
  * @element dds-masthead-menu-button
  */
 @customElement(`${ddsPrefix}-masthead-menu-button`)
-class DDSMastheadMenuButton extends BXHeaderMenuButton {
+class DDSMastheadMenuButton extends HostListenerMixin(BXHeaderMenuButton) {
   /**
    * The handle for focus wrapping.
    */
   private _hFocusWrap: Handle | null = null;
 
+  /**
+   * Media query listener handler.
+   */
+  private _mqHandler;
+
+  /**
+   * Search bar opened flag.
+   */
+  @internalProperty()
+  private _openedSearch = false;
+
+  /**
+   * Medium breakpoint flag.
+   */
+  @internalProperty()
+  private _responsiveMd = false;
+
+  /**
+   * Handles toggle event from the search component.
+   *
+   * @param event The event.
+   */
+  @HostListener(`parentRoot:${DDSMastheadSearch.eventToggle}`)
```

Comment:
To support alternate (typically) component composed in masthead, this codebase makes event name configurable in receiving component side ([an example](https://github.com/carbon-design-system/carbon-web-components/blob/v1.5.0/src/components/ui-shell/side-nav.ts#L198-L200)). So please define `eventToggle` static property to each classes receiving the event:

```suggestion
  @HostListener(`parentRoot:eventToggle`)
```